app/cdap/components/ColumnInsights/Components/ColumnDetails/index.tsx

```diff
@@ -0,0 +1,194 @@
+`;
+
+export default function({
+  columnName,
+  characterCount,
+  distinctValues,
+  dataTypeString,
+  renameColumnNameHandler,
+  dataTypeHandler,
+  columnType,
+  columnHeaderList,
+}: IColumnDetailsProps) {
+  const defaultValueProvided =
+    DATATYPE_OPTIONS &&
+    Array.isArray(DATATYPE_OPTIONS) &&
+    DATATYPE_OPTIONS?.length &&
+    DATATYPE_OPTIONS.filter((each) => each.value === columnType?.toLowerCase());
+  const [dataTypeValue, setDataTypeValue] = useState<string>();
+  const [canEdit, setCanEdit] = useState(false);
+  const [inputValue, setInputValue] = useState(columnName);
+  const [errorMessage, setErrorMessage] = useState({
+    hasError: false,
+    message: '',
+  });
+
+  const displayMessage = {
+    invalidMessage: `${PREFIX}.error.invalidError`,
+    matchedColumnMessage: `${PREFIX}.error.matchedColumnError`,
+  };
+
+  const checkForInvalidInput = (renamedString: string) => {
+    if (renamedString === '' || !/^\w+$/.test(renamedString)) {
+      setErrorMessage({
+        hasError: true,
+        message: displayMessage.invalidMessage,
+      });
+    } else if (columnHeaderList.includes(renamedString) && columnName !== renamedString) {
+      setErrorMessage({
+        hasError: true,
+        message: displayMessage.matchedColumnMessage,
+      });
+    } else {
+      setErrorMessage({
+        hasError: false,
+        message: '',
+      });
+    }
+  };
+
+  const handleDataTypeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDataTypeValue(e.target.value);
+    dataTypeHandler(e.target.value);
+  };
+
+  const editHandler = () => {
+    setCanEdit(true);
+  };
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    checkForInvalidInput(e.target.value);
+    setInputValue(e.target.value);
+  };
+
+  const onEnter = (e: React.KeyboardEvent<HTMLElement>) => {
+    if (
+      (e.target as HTMLInputElement).value !== columnName &&
+      !errorMessage.hasError &&
+      e.keyCode === 13
+    ) {
+      renameColumnNameHandler(columnName, (e.target as HTMLInputElement).value);
+      setCanEdit(false);
+    } else {
+      setCanEdit(true);
+    }
+  };
+
+  return (
+    <ColumnDetailsContainer data-testid="column-details-parent">
+      <CustomizedColumnNameEditBox>
+        {canEdit ? (
+          <input
+            value={inputValue}
+            onChange={(e) => onChangeHandler(e)}
+            onKeyDown={(e) => onEnter(e)}
+            data-testid="column-name-edit-input"
+            autoFocus
+          />
+        ) : (
+          <RenderLabel fontSize={16} dataTestId={'column-name'}>
+            <> {inputValue}</>
+          </RenderLabel>
+        )}
+        <CustomizedIconButton onClick={editHandler} aria-label="edit-icon" data-testid="edit-icon">
+          <EditIcon />
+        </CustomizedIconButton>
+      </CustomizedColumnNameEditBox>
+      {errorMessage.hasError && (
+        <RenderLabel fontSize={14} color={`${red[600]}`} dataTestId={'invalid-text'}>
```

Comment:
I'm glad we're showing the error inline, but please see https://v4.mui.com/components/text-fields/#validation on how to display an error with a Mui `TextField`.

Reply:
Used Textfield to display Error.